lexmapr/pipeline_classification.py

```diff
@@ -133,14 +248,80 @@ def refine_ifsac_final_labels(sample, ifsac_final_labels, label_refinements):
     if "plant" in ret and ret.intersection(plant_categories):
         ret.remove("plant")
 
-    if "food" in ret and ret.intersection(animal_categories | plant_categories | {"meat"}):
-            ret.remove("food")
+    if "nut" in ret and "seeds" in ret and len(ret) == 2:
+         ret.remove("seeds")
+    if "environment-water" in ret and "fish" in ret and len(ret) == 2:
+         ret.remove("environment-water")
 
+    # Retains the specific (more granular) environmental classes
+    if "environmental" in ret and ret.intersection(environmental_categories):
+        ret.remove("environmental")
+    if ("environmental-animal housing" in ret or "environmental-abattoir" in ret
+            or "environmental-farm" in ret) \
+            and "environmental-factory/production facility" in ret:
+        ret.remove("environmental-factory/production facility")
+    if "environmental-abattoir" in ret and "environmental-factory/production facility" in ret:
+        ret.remove("environmental-factory/production facility")
+
+    if not ("clinical/research" in ret or "veterinary clinical/research" in ret or "animal feed" in ret
+            or "human" in ret or "environmental" in ret or ret.intersection(environmental_categories)) \
+            and len(ret) >= 3:
```

Comment:
```suggestion
    exclusions = {
        'clinical/research', 'veterinary clinical/research', 'animal feed', 'human',
        'environmental'
    }
    if not (ret.intersection(exclusions) or ret.intersection(environmental_categories)) \
            and len(ret) >= 3:
```

Line is too long, and kind of messy. You can do things like this to clean it up.